bundles/org.openhab.binding.shelly/src/main/resources/ESH-INF/thing/sensor.xml

```diff
@@ -123,4 +102,121 @@
 		</channels>
 	</channel-group-type>
 
+
+	<channel-type id="sensorState">
+		<item-type>Contact</item-type>
+		<label>Contact</label>
+		<description>State of the contact: open/close</description>
+		<category>Contact</category>
+		<state pattern="%s" readOnly="true">
+			<options>
+				<option value="OPEN">Open</option>
+				<option value="CLOSED">Closed</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="sensorWakeup" advanced="true">
+		<item-type>String</item-type>
+		<label>Wakeup reason</label>
+		<description>Reasons which woke up the device: battery, button, periodic, poweron, sensor, alarm</description>
+		<state readOnly="true">
+		</state>
+	</channel-type>
+
+	<channel-type id="sensorTemp">
+		<item-type>Number:Temperature</item-type>
+		<label>Temperature</label>
+		<description>Temperature from the sensor</description>
+		<category>Temperature</category>
+		<tags>
+			<tag>CurrentTemperature</tag>
+		</tags>
+		<state readOnly="true" pattern="%f %unit%">
+		</state>
+	</channel-type>
+	<channel-type id="sensorExtTemp">
+		<item-type>Number:Temperature</item-type>
+		<label>Temperature</label>
+		<description>Temperature from the external sensor</description>
+		<category>Temperature</category>
+		<tags>
+			<tag>CurrentTemperature</tag>
+		</tags>
+		<state readOnly="true" pattern="%f %unit%">
+		</state>
+	</channel-type>
+	<channel-type id="sensorExtHum">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Humidity</label>
+		<description>Relative humidity in % (0..100%) from external sensor</description>
+		<category>Humidity</category>
+		<tags>
+			<tag>CurrentHumidity</tag>
+		</tags>
+		<state readOnly="true" pattern="%f %unit%">
+		</state>
+	</channel-type>
+	<channel-type id="sensorHumidity">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Humidity</label>
+		<description>Relative humidity in % (0..100%)</description>
+		<tags>
+			<tag>CurrentHumidity</tag>
+		</tags>
+		<state readOnly="true" min="0" max="100" pattern="%f %unit%" />
+	</channel-type>
+	<channel-type id="sensorFlood">
+		<item-type>Switch</item-type>
```

Comment:
I wonder if this is more of a contact than a switch? 🤔

Reply:
hmm, I would say is not a Contact nore a Switch, it's a State (Alarm yes/no)
changing this means breaking backward compatibility and all things need to be re-discovered